pkg/cri/config/config.go

```diff
@@ -156,6 +156,9 @@ type Registry struct {
 	Auths map[string]AuthConfig `toml:"auths" json:"auths"`
 	// Headers adds additional HTTP headers that get sent to all registries
 	Headers map[string][]string `toml:"headers" json:"headers"`
+	// HostsDir sets the root directory for registry host configuration
+	// If a HostsDir is set, then registries other registry configs are ignored
+	HostsDir string `toml:"hosts_dir" json:"hosts_dir"`
```

Comment:
I think this should be defined as `[]string` to allow specifying multiple directories.
The actual implementation for multi-dir can be another PR.

Reply:
agreed!  ` hosts_dir = ["/etc/docker/certs.d", "/etc/containerd/cri/hosts"]`  would be great for people migrating from docker to containerd.

Reply:
sounds good to have more than one.. but may be confusing with multiple disparate config locations.. which has precedence.. do we error on overwrite. Just have to consider that complexity.

Reply:
I would argue against this for the sake of saving complexity and rather err on the admin to configure the system rather than trying to make containerd handle all possibilities.

Reply:
Maybe its not *that* bad if we add a helper function like `func(registryHosts ...RegistryHosts) RegistryHosts`.
Precedence is first match based on order of the `[]string`